web/api/backend.go

```diff
@@ -113,57 +121,77 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		// If the structure of the body is wrong, return an HTTP error.
 		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, "UNABLE TO PARSE JSON")
 		return
 	}
 
-	// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
-	if (requestBody.UseToken && len(requestBody.Token) == 0) || (len(requestBody.Credential) == 0) {
-		w.WriteHeader(http.StatusBadRequest)
+	// JSON BODY representing the credentials attribute to be used (i.e. either the credentials found in the token or inputted).
+	creds := make(map[string]interface{})
+
+	// JSON Body representing the credentials attribute with the following format: "credentials":{"scopes":[],...}
+	// This is so to match with how the wrapper class will save the credentials file to memory.
+	credsMap := make(map[string]interface{})
+
+	// credsMap needs to nil if command type is reset, test, or info. Otherwise, credsMap will be chnage in the conditional statement that follows.
+	credsMap = nil
+
+	// If command type is test or token, credentials are not necessary
+	if isCredentialsNeeded(requestBody.CommandType) {
+		// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
+		if (requestBody.UseToken && len(requestBody.Token) == 0) || (!requestBody.UseToken && len(requestBody.Credential) == 0) {
```

Comment:
In this case, can you move this check to L155 since you also check for requestBody.UseToken there.

Reply:
Are you asking to move the isCredentialNeeded from L139 to L155? Because this check is needed at line 139 since when a token is being validated, a credentials and token will not be present, and moving that check will result in a "missing..." error.

Reply:
Ah I see what you mean, this looks okay.